### DISCUSSION TITLE
Git Mob uses Git global config (v2.1.0)

### DISCUSSION THREAD

**Discussion by rkotze** [findmypast-oss/git-mob#81] [Announcements]
Selected co-authors are **stored globally** meaning when switching between projects your co-authors stay the same.

It's likely you will see a warning message because previous versions set a local commit template.

If you've set a **local** commit template in your config then that template will be updated as usual. However, **not** when you switch projects. You will see selected co-authors if you run `git mob` command but they will **not** be in your template. You can update the template by running the `git mob <author initials>` command.

If you're only using the template for Git Mob, then it's recommended to remove your local `commit.template` config and use the global config.

**How to migrate** to the global setup:

- Remove local template: `git config --local --unset commit.template` 
- Re-run `git mob <author initials>` to update to the global commit template.

This does **not affect** the `git mob-print` command as it reads the globally selected co-authors.

## I use the local commit template

In the situation, you are using a local commit template and want to suppress the warning message you can. See Git Mob config in the readme for more information. This change is released in version `2.2.0`.

CD into your project and run the following:

`git config --local git-mob-config.use-local-template true`

We **don't force** the use of a global template as this would not follow the priority order of [Git config](https://git-scm.com/docs/git-config#FILES).

**Comment by niksumeiko** [comment]
After migrating to the global setup, I don't see the co-author trailer is added to my commits. The `git mob` command outputs current co-authors, but when committing, nothing is appended to the commit.

**Reply by rkotze** [reply]
Sorry to hear that @niksumeiko 

Do you see any error message?

The key thing is to make sure the git config has a commit.template with a valid path.

**Reply by niksumeiko** [reply]
Thanks for the tip, @rkotze. The issue was indeed in the path to my commit.template. After migrating to using a local commit template by following the guidelines in this discussion description, the path was still referring to my global one:

```bash
$ git config --local --get commit.template
# ~/.gitmessage
```

But my husky `.husky/prepare-commit-msg` hook was using the local one:
```bash
file=.git/.gitmessage
```

Therefore, after running `git mob xy`, the trailer was written in my global commit.template, but not in the local one. And husky was not picking it (since it was looking at the contents of `.git/.gitmessage`).

**The fix was to set the path properly:**
```bash
git config --local commit.template .git/.gitmessage
```

After this, the `git mob xy` command was writing the trailer into the local commit.template and husky picked it properly 🎉

**Reply by rkotze** [reply]
Glad you got that resolved and thank you for sharing your learnings.
